factorio.nix

```diff
@@ -0,0 +1,8 @@
+{ ... }: {
+  nixpkgs.config.allowUnfree = true;
+  environment.persistence."/state".directories = [
+    "/var/lib/private/factorio"
```

Comment:
Where is this directory derived from? According to the `nixpkgs` definition for factorio (https://github.com/NixOS/nixpkgs/blob/e56990880811a451abd32515698c712788be5720/nixos/modules/services/games/factorio.nix#L118C5-L118C5) the default is `/var/lib/${config.services.factorio.stateDirName}`

Reply:
If you look at the systemd service definition:
```
        DynamicUser = true;
        StateDirectory = cfg.stateDirName;
```

And from the systemd manpage:
> If DynamicUser= is used, the logic for CacheDirectory=, LogsDirectory= and StateDirectory= is slightly altered: the directories are created below /var/cache/private, /var/log/private and /var/lib/private, respectively, which are host directories made inaccessible to unprivileged users, which ensures that access to these directories cannot be gained through dynamic user ID recycling. Symbolic links are created to hide this difference in behaviour. Both from perspective of the host and from inside the unit, the relevant directories hence always appear directly below /var/cache, /var/log and /var/lib.

(but I cheated and just found the directory myself :^)